Project-Euler/Problem035.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * Problem 35 - Circular primes
+ *
+ * @see {@link https://projecteuler.net/problem=35}
+ *
+ * The number, 197, is called a circular prime because all rotations of the digits: 197, 971, and 719, are themselves prime.
+ * There are thirteen such primes below 100: 2, 3, 5, 7, 11, 13, 17, 31, 37, 71, 73, 79, and 97.
+ * How many circular primes are there below one million?
+ *
+ * @author ddaniel27
+ */
+
+/**
+ * Function to get all prime numbers below a given number
+ * This function also discards all numbers that contain an even digit
+ * @see {@link https://en.wikipedia.org/wiki/Sieve_of_Eratosthenes}
+ */
+function getPrimes (max) {
+  const sieve = []
+  const primes = []
+
+  for (let i = 2; i <= max; ++i) {
+    if (!sieve[i]) { // If i has not been marked then it is prime
+      const hasEven = String(i).split('').some((n) => n % 2 === 0) // Check if i contains an even digit
```

Comment:
> I did an early filter because for this problem this gave me like +7700 numbers with even digits.

7700 is a pretty small number for computers to handle these days. If you want to minimize the overhead incurred, don't use a temporary array, but rather [`yield` from a generator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/yield). Then it won't make *any* difference whether the check is before or after the yield. Having the check *outside* of this function allows this to be a clean Sieve of Eratosthenes, easily swapped out with any other Sieve or imported from somewhere else.

The updated function name is still inaccurate: This discards primes containing two, not only even primes. `23` is not even, yet would be discarded by this. An accurate name would be `getMaybeCircularPrimes`, which is a weirder building block than just an unmodified sieve for getting primes.

Reply:
Something to add after doing some little tests in local, I wrote wrong the amount of numbers, is not +7700, the actual amount for PE35 is `78498` and with my sieve adaptation this number decrease to `3217`. This is a _huge_ decrement, so my solution can be optimized doing some checks of even digits after getting the prime numbers list.

Reply:
I've created another pull request to add my implementation of Sieve of Eratosthenes because I think it's more straightforward get the list of prime numbers than a large array full of boolean values. With that algorithm I'll update this PR again with the optimized version of my solution.